base_location_nuts/wizard/nuts_import.py

```diff
@@ -0,0 +1,223 @@
+#
+##############################################################################
+
+from openerp import models, api, _
+from openerp.exceptions import Warning
+import requests
+import re
+import logging
+from lxml import etree
+from collections import OrderedDict
+
+from pprint import pformat
+
+logger = logging.getLogger(__name__)
+
+
+class NutsImport(models.TransientModel):
+    _name = 'nuts.import'
+    _description = 'Import NUTS items from European RAMON service'
+    _parents = [False, False, False, False]
+    _countries = {
+        "BE": False,
+        "BG": False,
+        "CZ": False,
+        "DK": False,
+        "DE": False,
+        "EE": False,
+        "IE": False,
+        "GR": False,  # EL
+        "ES": False,
+        "FR": False,
+        "HR": False,
+        "IT": False,
+        "CY": False,
+        "LV": False,
+        "LT": False,
+        "LU": False,
+        "HU": False,
+        "MT": False,
+        "NL": False,
+        "AT": False,
+        "PL": False,
+        "PT": False,
+        "RO": False,
+        "SI": False,
+        "SK": False,
+        "FI": False,
+        "SE": False,
+        "GB": False,  # UK
+    }
+    _current_country = False
+    _map = OrderedDict([
+        ('level', {
+            'xpath': '', 'attrib': 'idLevel',
+            'type': 'integer', 'required': True}),
+        ('code', {
+            'xpath': './Label/LabelText[@language="ALL"]',
+            'type': 'string', 'required': True}),
+        ('name', {
+            'xpath': './Label/LabelText[@language="EN"]',
+            'type': 'string', 'required': True}),
+    ])
+
+    def _check_node(self, node):
+        if node.get('id') and node.get('idLevel'):
+            return True
+        return False
+
+    def _mapping(self, node):
+        item = {}
+        for k, v in self._map.iteritems():
+            field_xpath = v.get('xpath', '')
+            field_attrib = v.get('attrib', False)
+            field_type = v.get('type', 'string')
+            field_required = v.get('required', False)
+            value = ''
+            if field_xpath:
+                n = node.find(field_xpath)
+            else:
+                n = node
+            if n is not None:
+                if field_attrib:
+                    value = n.get(field_attrib, '')
+                else:
+                    value = n.text
+                if field_type == 'integer':
+                    try:
+                        value = int(value)
+                    except:
+                        value = 0
+            else:
+                logger.debug("xpath = '%s', not found" % field_xpath)
+            if field_required and not value:
+                raise Warning(
+                    _('Value not found for mandatory field %s' % k))
+            item[k] = value
+        return item
+
+    def _download_nuts(self):
+        url_base = 'http://ec.europa.eu'
+        url_path = '/eurostat/ramon/nomenclatures/index.cfm'
```

Comment:
1) URLs are data, storing data in code is bad design
2) there is no guarantee that that url never changes
3) maybe you don't want load data from external server, so you cache the data somewhere on your server and then load the data to odoo, then you have also a different url (in the most case this depends on it policy in a organisation)